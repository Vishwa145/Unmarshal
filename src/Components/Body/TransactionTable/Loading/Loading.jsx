import React, {useEffect, useState} from 'react';

function Loading (){
    const [loading, changeLoading] = useState("...");

    useEffect(()=>{
        let Timer_id = setInterval(()=> {
            changeLoading((prev)=>prev.length===3?"":prev+".");
        }, 500);
        return ()=>{
            clearInterval(Timer_id);
        }
    },[]);
    return(
        <h1>Loading{loading}</h1>
    );
}

export default Loading;
