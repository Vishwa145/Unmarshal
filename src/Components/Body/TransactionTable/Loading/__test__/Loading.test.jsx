import React from "react";
import ReactDOM from "react-dom";
import Loading from "../Loading";
import renderer from "react-test-renderer"

it("render successfully", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Loading />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("Matches snapshot", ()=>{
    const tree = renderer.create(<Loading />).toJSON();
    expect(tree).toMatchSnapshot();
});