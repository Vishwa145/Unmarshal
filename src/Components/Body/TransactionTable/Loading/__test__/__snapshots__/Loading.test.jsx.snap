// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<h1>
  Loading
  ...
</h1>
`;
